src/xas_tdp_atom.F

```diff
@@ -1524,20 +1552,34 @@ SUBROUTINE put_density_on_other_grid(rho_set, ri_dcoeff, source_iat, source_ikin
 !$OMP DO COLLAPSE(2) SCHEDULE(STATIC)
                      DO ir = sr, er
                         DO ia = 1, na
-                           dco(ia, ir, 3, start + ico) = -2.0_dp*pos(ia, ir, 3)*zet(ipgf, iset) &
-                                                         *pos(ia, ir, 1)**lx*pos(ia, ir, 2)**ly &
-                                                         *EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
+                           rmom = -2.0_dp*pos(ia, ir, 3)*zet(ipgf, iset)*EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
+                           IF (lx /= 0) rmom = rmom*pos(ia, ir, 1)**lx
+                           IF (ly /= 0) rmom = rmom*pos(ia, ir, 2)**ly
+                           dco(ia, ir, 3, start + ico) = rmom
+!                          dco(ia, ir, 3, start + ico) = -2.0_dp*pos(ia, ir, 3)*zet(ipgf, iset) &
+!                                                        *pos(ia, ir, 1)**lx*pos(ia, ir, 2)**ly &
+!                                                        *EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
                         END DO
                      END DO
 !$OMP END DO NOWAIT
                   ELSE
 !$OMP DO COLLAPSE(2) SCHEDULE(STATIC)
                      DO ir = sr, er
                         DO ia = 1, na
-                           dco(ia, ir, 3, start + ico) = (lz*pos(ia, ir, 3)**(lz - 1) &
-                                                          - 2.0_dp*pos(ia, ir, 3)**(lz + 1)*zet(ipgf, iset)) &
-                                                         *pos(ia, ir, 1)**lx*pos(ia, ir, 2)**ly &
-                                                         *EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
+                           IF (lz /= 1) THEN
+                              rmom = lz*pos(ia, ir, 3)**(lz - 1) - 2.0_dp*pos(ia, ir, 3)**(lz + 1)* &
+                                     zet(ipgf, iset)*EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
+                           ELSE
+                              rmom = (1.0_dp - 2.0_dp*pos(ia, ir, 3)**2)*zet(ipgf, iset)* &
+                                     EXP(-zet(ipgf, iset)*pos(ia, ir, 4))
```

Comment:
Here, the closing parenthesis is misplaced, it should be (see bold parenthesis, moved after zet):

<code>rmom = (1.0_dp - 2.0_dp *pos(ia, ir, 3)**2 *zet(ipgf, iset)__)__ * EXP(-zet(ipgf, iset)*pos(ia, ir, 4))</code>